src/platform/silabs/EndpointQueueFilter.cpp

```diff
@@ -0,0 +1,141 @@
+#include "EndpointQueueFilter.h"
+#include <string.h>
+#include <support/CodeUtils.h>
+#include <support/logging/CHIPLogging.h>
+
+namespace chip {
+namespace Inet {
+
+using FilterOutcome = EndpointQueueFilter::FilterOutcome;
+using namespace chip::Logging;
+
+namespace {
+
+bool IsValidMdnsHostName(const chip::CharSpan & hostName)
+{
+    for (size_t i = 0; i < hostName.size(); ++i)
+    {
+        char ch_data = *(hostName.data() + i);
+        if (!((ch_data >= '0' && ch_data <= '9') || (ch_data >= 'A' && ch_data <= 'F')))
+        {
+            return false;
+        }
+    }
+    return true;
+}
+
+bool PayloadContains(const chip::System::PacketBufferHandle & payload, const chip::ByteSpan & byteSpan)
+{
+    if (payload->HasChainedBuffer() || payload->TotalLength() < byteSpan.size())
+    {
+        return false;
+    }
+    for (size_t i = 0; i <= payload->TotalLength() - byteSpan.size(); ++i)
```

Comment:
Any chance this is the same:

```cpp
std::basic_string_view<uint8_t> needle(byteSpan.data(), byteSpan.size());
std::basic_string_view<uint8_t> haystack(payload->Start(), payload->TotalLength);

return haystack.find(needle) != std::npos;
```

(note: not tested code, however figured maybe shorter/clearer).

Reply:
Thanks for the suggestion, I have verified it. It works. Thanks!